main.go

```diff
@@ -215,3 +221,30 @@ func ensureManagedOCS(c client.Client, log logr.Logger, envVars map[string]strin
 		return err
 	}
 }
+
+func checkIfCRDExist(crdNames []string) map[string]bool {
+
+	var options client.Options
+	options.Scheme = runtime.NewScheme()
+	utilruntime.Must(apiextensionsv1.AddToScheme(options.Scheme))
+
+	k8sClient, err := client.New(config.GetConfigOrDie(), options)
+	if err != nil {
+		setupLog.Error(err, "error creating client")
+		os.Exit(1)
+	}
+
+	CRDExist := map[string]bool{}
```

Comment:
- a teeny tiny optimisation here
```suggestion
	CRDExist := map[string]struct{}
```
- and change
``` go
// set op
// from
CRDExist[crdName] = crd.UID != ""

// to
if crd.UID != "" {
  CRDExist[crdName] = struct{}{}
}

// get op
// from
if CRDExist[crdName] {}

// to
if _, exist := CRDExist[crdName]; exist {}
```
- allocation of `struct{}` costs you literally nothing compared to any type
- if you don't feel readability is going to take a hit above change can be made

Reply:
I think this is a no-gain. 
The common practice is to have a map of bools to implement a set. To enable removing from the set without the need to actually change anything in memory